### PR TITLE
CON-2459: extended addProductsToCart request to also include option data

### PR DESCRIPTION
**Pull request by kernelkiller**

### CODE REVIEW

File: product/reducers/optionsByProductId.js [hunk 1]
```diff
@@ -29,16 +30,42 @@ export default function optionsByProductId(state = {}, action) {
           expires: 0,
         },
       };
-    case RECEIVE_PRODUCT_OPTIONS:
+    case RECEIVE_PRODUCT_OPTIONS: {
+      const options = action.options.map((option) => {
```
Comment (richardgorman) [inline product/reducers/optionsByProductId.js]: I think that this formatting code should be moved into a separate helper file.

File: cart/selectors/index.js [hunk 2]
```diff
@@ -167,3 +179,53 @@ export const getCartMessages = createSelector(
...
+  areProductOptionsSet,
+  getRawProductOptions,
+  getCurrentProductOptions,
+  (hasOptions, areOptionsSet, options, currentOptions) => {
+    // Check if options are ready to be added to a pipeline request.
+    if (!hasOptions || !areOptionsSet) {
+      return null;
+    }
+
+    // Create the datastructure
```
Comment (richardgorman) [inline cart/selectors/index.js]: // Create the data structure.

File: cart/selectors/index.js [hunk 3]
```diff
@@ -167,3 +179,53 @@ export const getCartMessages = createSelector(
...
+      // Use the metadata from the selected product if available.
+      return metaData;
+    }
+
+    return null;
+  }
+);
+
+/**
+ * Creates data for the "options" property of addProductsToCart pipeline request payload.
```
Comment (richardgorman) [inline cart/selectors/index.js]: Builds the data for the "options" property of addProductsToCart pipeline request payload.

File: cart/selectors/index.js [hunk 4]
```diff
@@ -167,3 +179,53 @@ export const getCartMessages = createSelector(
   getCart,
   cart => cart.messages
 );
+
+/**
+ * Creates data for the "metadata" property of addProductsToCart pipeline request payload.
```
Comment (richardgorman) [inline cart/selectors/index.js]: Builds the data for the 'metadata' property of addProductsToCart pipeline request payload.

File: product/reducers/optionsByProductId.js (hunk 1 above)
Comment (Carsten89) [reply product/reducers/optionsByProductId.js]: done

File: cart/selectors/index.js (hunk 4 above)
Comment (Carsten89) [reply cart/selectors/index.js]: done

File: cart/selectors/index.js (hunk 3 above)
Comment (Carsten89) [reply cart/selectors/index.js]: done

File: cart/selectors/index.js (hunk 2 above)
Comment (Carsten89) [reply cart/selectors/index.js]: done

**Pull request merged**